jxl/src/frame/modular/transforms/apply.rs

```diff
@@ -132,24 +148,112 @@ impl TransformStepChunk {
                     let img_pal = Ref::map(buffers[*buf_pal].buffer_grid[0].data.borrow(), |x| {
                         x.as_ref().unwrap()
                     });
-                    with_buffers(buffers, buf_out, out_grid, |mut bufs| {
-                        super::palette::do_palette_step_general(
-                            &img_in,
-                            &img_pal,
-                            &mut bufs,
-                            *num_colors,
-                            *num_deltas,
-                            *predictor,
-                            wp_header,
-                        );
-                        Ok(())
-                    })?;
+                    with_buffers(buffers, buf_out, out_grid, |_| Ok(()))?;
```

Comment:
Is this just meant to ensure that the buffers are present?

I am guessing you're not doing the thing in the callback, and I imagine that's because the interface is not powerful enough, but it might be nicer to extend that code instead

Reply:
Yes, it is just for creating buffers. Added a comment and a TODO for extending the callback functionality.